### PR TITLE
GUM-1: Login screen

### PR DESCRIPTION
**Pull request by jol79**

### CODE REVIEW

File: src/components/input/theme.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const theme = {};
```
Comment (AleksandrTermenzhy) [inline src/components/input/theme.ts]: Why do we need this?

File: src/screens/Login/Components/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { LoginForm } from './login-form';
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/Components/index.ts]: Please stick to existing folder naming style (lowercase)

File: src/screens/Login/Components/login-form.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { Button, TextInput } from 'react-native-paper';
+import { View, Image } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+
+import { styles } from '../styles';
+import { Input, Text } from 'src/components';
+import { AppColor, AuthRoute } from 'src/common/enums';
+
+const LoginForm = () => {
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/Components/login-form.tsx]: As long as this component is Form, let's keep only form related stuff here. And move logo and footer to the screen-component.

Comment (AleksandrTermenzhy) [inline src/screens/Login/Components/login-form.tsx]:
```suggestion
const LoginForm: React.FC = () => {
```

File: src/screens/Login/Components/login-form.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        outlineColor={AppColor.INPUT_BACKGROUND}
+        label="Password"
+        value={password}
+        secureTextEntry={secure}
+        onChangeText={text => setPassword(text)}
+        right={
+          <TextInput.Icon
+            name="eye"
+            style={styles.formIcon}
+            onPress={() => setSecure(!secure)}
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/Components/login-form.tsx]: Would be better to introduce a separate toggle function

File: src/screens/Login/Components/login-form.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          />
+        }
+      />
+
+      <Button mode="contained" style={styles.btnLogin} onPress={onLoginPressed}>
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/Components/login-form.tsx]: I believe we have `MainButton` component for this.

File: src/screens/Login/login.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { SafeAreaView } from 'react-native';
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/login.tsx]:
```suggestion
import { SafeAreaView } from 'react-native-safe-area-context';
```

Actually it's a good point to re-export `SafeAreaView` in `components/index.ts` to avoid such errors in future

File: src/screens/Login/login.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import { LoginForm } from 'src/screens/Login/components';
+
+type ILoginScreenProps = Record<string, never>;
+
+const LoginScreen: React.FC<ILoginScreenProps> = () => {
+  return (
+    <SafeAreaView>
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/login.tsx]: There is a good practice to add `flex: 1` to root component

File: src/screens/Login/styles.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
+import { StyleSheet } from 'react-native';
+
+import { AppColor } from 'src/common/enums';
+
+const styles = StyleSheet.create({
+  container: {
+    marginHorizontal: 27,
+    marginVertical: 22,
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/styles.ts]: What are those magic numbers? We might already have default screen padding shared constant somewhere in code. Ask @katyakostikova for more details on this

File: src/screens/Login/styles.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  formFieldErrorMessage: {
+    color: AppColor.ACCENT,
+  },
+  btnLogin: {
+    borderRadius: 50,
+    backgroundColor: AppColor.ACCENT,
+    height: 45,
+    justifyContent: 'center',
+    marginTop: '45%',
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/styles.ts]: To stick button to the bottom you may do this way:
```
<Root style={{ flex: 1 }}>
    <FormContainer style={{ flex: 1 }} />
   <ButtonContainer />
</Root>
```

File: src/screens/Login/styles.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  logo: {
+    width: 163,
+    height: 48,
+    alignSelf: 'center',
+    marginTop: '20%',
+    marginVertical: '20%',
```
Comment (AleksandrTermenzhy) [inline src/screens/Login/styles.ts]: `marginVertical` already includes `marginTop`

**Comment by AleksandrTermenzhy** [comment]
Please follow existing pull-request naming: `GUM-<ticket_number>: short description`

File: src/screens/Login/styles.ts (hunk 8 above)
Comment (katyakostikova) [reply src/screens/Login/styles.ts]: Yes, I've added this in src/styles but it's still in PR #27

File: src/screens/Login/Components/login-form.tsx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  const navigation = useNavigation();
+
+  const onLoginPressed = () => {
+    // TODO authentication logic
+  };
+
+  return (
+    <View style={styles.container}>
+      <Image
+        source={require('src/assets/images/Logo.png')}
```
Comment (R-Mielamud) [inline src/screens/Login/Components/login-form.tsx]: In .tsx this is different. Add this line before styles import:
```typescript
import logoImage from 'src/assets/images/Logo.png'
```
And then `source={logoImage}`

File: src/screens/Login/Components/login-form.tsx [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+      </Button>
+
+      <View style={styles.footer}>
+        <Text>New to GrowUp?</Text>
+        <Text
+          style={styles.footerRedirect}
+          onPress={() => navigation.navigate(AuthRoute.SIGN_UP)}
+        >
+          Sign Up
+        </Text>
```
Comment (R-Mielamud) [inline src/screens/Login/Components/login-form.tsx]: And for this we use `mode={ButtonMode.TEXT}`
```suggestion
        <MainButton
          mode={ButtonMode.TEXT}
          style={styles.footerRedirect}
          onPress={() => navigation.navigate(AuthRoute.SIGN_UP)}
        >
          Sign Up
        </MainButton>
```

File: src/screens/register/register.tsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+type IRegisterScreenProps = Record<string, never>;
+
+const RegisterScreen: React.FC<IRegisterScreenProps> = () => {
+  const navigation = useNavigation();
+
+  return (
+    <SafeAreaView>
+      <View>
+        <Text>Register screen</Text>
+        <Text onPress={() => navigation.goBack()}>Sign in</Text>
```
Comment (R-Mielamud) [inline src/screens/register/register.tsx]: Also mode: TEXT

File: src/screens/Login/Components/index.ts (hunk 2 above)
Comment (jol79) [reply src/screens/Login/Components/index.ts]: image

Comment (R-Mielamud) [reply src/screens/Login/Components/index.ts]: Known problem :)
Steps to fix:
1. Rename folders back to Login/Components
2. Delete them or move to gitignored folder
3. `git add .`
4. Return folders back
5. Rename to kebab-case
6. `git add .`
7. Commit, push, pull request, etc.

File: src/screens/Login/Components/login-form.tsx (hunk 11 above)
Comment (jol79) [reply src/screens/Login/Components/login-form.tsx]: It doesn't work for me
image

I can create a separate file with exports of the images from assets.

File: src/components/shared/input-style/theme.ts [hunk 14]
```diff
@@ -2,7 +2,6 @@ import theme from 'src/styles/theme';
 
 const inputTheme: ReactNativePaper.Theme = {
   ...theme,
-  roundness: 0,
 };
 
 export default inputTheme;
```
Comment (katyakostikova) [inline src/components/shared/input-style/theme.ts]: You can remove this file

File: src/components/input/input.tsx [hunk 15]
```diff
@@ -19,7 +19,8 @@ const Input: React.FC<InputProps> = ({
       label={label}
       placeholder={placeholder}
       theme={theme}
```
Comment (katyakostikova) [inline src/components/input/input.tsx]:
```suggestion
```

File: src/screens/login/login.tsx [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+          style={styles.logo}
+        />
+        <LoginForm />
+      </View>
+
+      <View style={styles.footer}>
+        <Text>New to GrowUp?</Text>
+        <MainButton
+          mode={ButtonMode.TEXT}
+          onPress={() => navigation.navigate(AuthRoute.SIGN_UP)}
```
Comment (katyakostikova) [inline src/screens/login/login.tsx]: Maybe it'll be better to move it in a function?

File: src/screens/register/register.tsx [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+const RegisterScreen: React.FC<IRegisterScreenProps> = () => {
+  const navigation = useNavigation();
+
+  return (
+    <SafeAreaView>
+      <View>
+        <Text>Register screen</Text>
+        <MainButton
+          mode={ButtonMode.TEXT}
+          onPress={() => navigation.navigate(AuthRoute.SIGN_IN)}
```
Comment (katyakostikova) [inline src/screens/register/register.tsx]: same here about function

File: src/screens/login/components/login-form/login-form.tsx [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+
+  return (
+    <View style={styles.content}>
+      <View style={styles.container}>
+        <Input
+          style={styles.formField}
+          outlineColor={AppColor.INPUT_BACKGROUND}
+          label="Email"
+          value={email}
+          onChangeText={text => setEmail(text)}
```
Comment (katyakostikova) [inline src/screens/login/components/login-form/login-form.tsx]:
```suggestion
          onChangeText={setEmail)}
```

Comment (katyakostikova) [reply src/screens/login/components/login-form/login-form.tsx]: Same in password

File: src/screens/login/login.tsx [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+        />
+        <LoginForm />
+      </View>
+
+      <View style={styles.footer}>
+        <Text>New to GrowUp?</Text>
+        <MainButton
+          mode={ButtonMode.TEXT}
+          onPress={() => onNavigateTo(AuthRoute.SIGN_UP)}
+        >
```
Comment (katyakostikova) [inline src/screens/login/login.tsx]:
```suggestion
  const onNavigateTo = () => {
    navigation.navigate(AuthRoute.SIGN_UP);
  };

  return (
    <SafeAreaView style={styles.content}>
      <View style={styles.content}>
        <Image
          source={require('src/assets/images/Logo.png')}
          style={styles.logo}
        />
        <LoginForm />
      </View>

      <View style={styles.footer}>
        <Text>New to GrowUp?</Text>
        <MainButton
          mode={ButtonMode.TEXT}
          onPress={onNavigateTo}
        >
```

Comment (katyakostikova) [reply src/screens/login/login.tsx]: Same in register

File: src/components/shared/input-style/theme.ts [hunk 20]
```diff
@@ -1,8 +0,0 @@
-import theme from 'src/styles/theme';
```
Comment (R-Mielamud) [inline src/components/shared/input-style/theme.ts:1]: Could you explain, please, why did you delete this?

File: src/screens/login/components/login-form/login-form.tsx [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+import { FormInput } from 'src/components';
+import { IAuthenticateUser } from 'src/common/types';
+
+const LoginForm: React.FC = () => {
+  const [secure, setSecure] = useState(true);
+
+  const onLoginPressed = (values: IAuthenticateUser) => {
+    // TODO authentication logic
+    console.log('Received data from the form: ', values);
+  };
```
Comment (R-Mielamud) [inline src/screens/login/components/login-form/login-form.tsx]: Add further layers: redux and service. This TODO must be placed instead of http request.

File: src/screens/login/components/login-form/login-form.tsx [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+              secureTextEntry={secure}
+              right={
+                <TextInput.Icon
+                  name="eye"
+                  style={styles.formIcon}
+                  onPress={onChangePasswordSecure}
+                />
+              }
+            />
+          </View>
```
Comment (R-Mielamud) [inline src/screens/login/components/login-form/login-form.tsx]: Extract separate FormPasswordInput component.

File: src/screens/login/login.tsx [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { View, Image, Text } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+
+import { LoginForm } from 'src/screens/login/components';
+import { styles } from './styles';
+import { AppRoute, AuthRoute, ButtonMode } from 'src/common/enums';
+import { MainButton } from 'src/components';
+
+type ILoginScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/login/login.tsx]: Remove this

File: src/screens/login/login.tsx [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+
+const LoginScreen: React.FC<ILoginScreenProps> = () => {
```
Comment (R-Mielamud) [inline src/screens/login/login.tsx]: Just `React.FC = () => ...`

File: src/screens/login/login.tsx [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+  const navigation = useNavigation();
+
+  const onNavigateTo = (route: AuthRoute | AppRoute) => {
```
Comment (R-Mielamud) [inline src/screens/login/login.tsx]: `const navigateTo` or `const handleNavigateTo`

File: src/screens/login/styles.ts [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  formFieldErrorMessage: {
+    color: AppColor.ACCENT,
+  },
+  btnLogin: {
+    marginHorizontal: 27,
+  },
+  logo: {
+    width: 163,
+    height: 48,
```
Comment (R-Mielamud) [inline src/screens/login/styles.ts]: These are magic values. Solution - remove height prop.

File: src/screens/register/register.tsx [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <SafeAreaView>
+      <View>
+        <Text>Register screen</Text>
+        <MainButton
+          mode={ButtonMode.TEXT}
+          onPress={() => onNavigateTo(AuthRoute.SIGN_IN)}
+        >
+          Sign in
```
Comment (R-Mielamud) [inline src/screens/register/register.tsx]: Sign up

File: src/screens/register/register.tsx [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+
+import { MainButton } from 'src/components';
+import { AuthRoute, ButtonMode } from 'src/common/enums';
+
+type IRegisterScreenProps = Record<string, never>;
+
+const RegisterScreen: React.FC<IRegisterScreenProps> = () => {
+  const navigation = useNavigation();
+
+  const onNavigateTo = (route: AuthRoute) => {
```
Comment (R-Mielamud) [inline src/screens/register/register.tsx]: `const navigateTo` or `const handleNavigateTo`

File: src/validation-schemas/login/login.ts [hunk 29]
```diff
@@ -0,0 +1,17 @@
+import * as yup from 'yup';
+
+const authenticateUserValidationSchema = yup.object({
+  email: yup
+    .string()
+    .required()
+    .min(8, 'Email is too short')
+    .max(50, 'Email is too long')
+    .email('Invalid email format'),
```
Comment (R-Mielamud) [inline src/validation-schemas/login/login.ts]: Must be a valid email

File: src/validation-schemas/login/login.ts [hunk 30]
```diff
@@ -0,0 +1,17 @@
+import * as yup from 'yup';
+
+const authenticateUserValidationSchema = yup.object({
+  email: yup
+    .string()
+    .required()
+    .min(8, 'Email is too short')
+    .max(50, 'Email is too long')
```
Comment (R-Mielamud) [inline src/validation-schemas/login/login.ts]: Email must be from 8 to 50 characters

File: src/validation-schemas/login/login.ts [hunk 31]
```diff
@@ -0,0 +1,17 @@
...
+    .email('Invalid email format'),
+  password: yup
+    .string()
+    .required()
+    .min(8, 'Password is too short')
+    .max(50, 'Password is too long'),
```
Comment (R-Mielamud) [inline src/validation-schemas/login/login.ts]: Password must be from 8 to 50 characters

File: src/validation-schemas/login/login.ts (hunk 29 above)
Comment (jol79) [reply src/validation-schemas/login/login.ts]: Do you mean to change the validation error message for the email field?

File: src/screens/login/styles.ts (hunk 26 above)
Comment (jol79) [reply src/screens/login/styles.ts]: These are the size of the logo image, here's the result of removing the height property. If there's any better solution I would appreciate your advice.
image

File: src/validation-schemas/login/login.ts (hunk 29 above)
Comment (R-Mielamud) [reply src/validation-schemas/login/login.ts]: Yes

File: src/components/shared/input-style/theme.ts (hunk 20 above)
Comment (jol79) [reply src/components/shared/input-style/theme.ts:1]: image

File: src/screens/login/styles.ts (hunk 26 above)
Comment (R-Mielamud) [reply src/screens/login/styles.ts]: Add to style: `resizeMode: 'contain'`

Comment (R-Mielamud) [reply src/screens/login/styles.ts]: https://stackoverflow.com/questions/29476165/image-resizing-in-react-native#answer-39910052

Comment (jol79) [reply src/screens/login/styles.ts]: How can I avoid these huge margins?
image

Comment (R-Mielamud) [reply src/screens/login/styles.ts]: Then remove `width`, set `height` and `resizeMode: 'contain'`

**Review by katyakostikova — APPROVED** [review]

File: src/common/types/auth/auth-response.ts [hunk 32]
```diff
@@ -1,7 +1,7 @@
 import { IUser } from '..';
 
 type AuthResponse = {
-  accessToken: string;
+  token: string;
   user: IUser;
```
Comment (Liza-Veis) [inline src/common/types/auth/auth-response.ts:5]:
```suggestion
```

File: src/common/types/user/authenticate-user.interface.ts [hunk 33]
```diff
@@ -0,0 +1,6 @@
+interface IAuthenticateUser {
```
Comment (liza-veis) [inline src/common/types/user/authenticate-user.interface.ts]: Remove this, we already have ISignInPayload

File: src/screens/login/components/login-form/login-form.tsx [hunk 34]
```diff
@@ -0,0 +1,72 @@
...
+import { Formik } from 'formik';
+
+import { styles } from '../../styles';
+import { MainButton } from 'src/components';
+import { defaultLoginPayload } from '../../common';
+import { authenticateUserValidationSchema } from 'src/validation-schemas';
+import { FormInput, FormPasswordInput } from 'src/components';
+import { IAuthenticateUser } from 'src/common/types';
+import { useAppDispatch } from 'src/hooks';
+import { setCurrentUser, signIn } from 'src/store/auth/actions';
```
Comment (Liza-Veis) [inline src/screens/login/components/login-form/login-form.tsx:12]:
```suggestion
import { authActions } from 'src/store/actions';
```

File: src/screens/login/components/login-form/login-form.tsx [hunk 35]
```diff
@@ -0,0 +1,72 @@
...
+      .catch((err: Error) => {
+        // Use react alers/toaster notifications
+        Alert.alert(`${err.message}`);
+      });
+    handleAuthorizedUserData()
+      .unwrap()
+      .catch((err: Error) => {
+        Alert.alert(`${err.message}`);
+      });
+  };
```
Comment (liza-veis) [inline src/screens/login/components/login-form/login-form.tsx]: We don't need useCallback here
```suggestion
  const handleLogin = (values: ISignInPayload) => {
    dispatch(authActions.signIn(values))
      .unwrap()
      .catch((err: Error) => {
        // Use react alers/toaster notifications
        Alert.alert(`${err.message}`);
      });
  };
```

File: src/screens/login/components/login-form/login-form.tsx [hunk 36]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <Formik
+      initialValues={defaultLoginPayload}
+      validationSchema={authenticateUserValidationSchema}
+      validateOnMount={true}
+      onSubmit={handleLoginPressed}
```
Comment (liza-veis) [inline src/screens/login/components/login-form/login-form.tsx]:
```suggestion
      onSubmit={handleLogin}
```

File: src/services/http/index.ts [hunk 37]
```diff
@@ -57,8 +57,11 @@ class Http {
     }
 
     if (hasAuth) {
+      console.log('TOKEN REQUIRED!');
       const token = await this.#storage.getItem(SecureStorageKey.ACCESS_TOKEN);
-      headers.append(HttpHeader.AUTHORIZATION, `Bearer ${token}`);
+      console.log('TOKEN: ', token);
+
+      headers.append(HttpHeader.AUTHORIZATION, `${token}`);
```
Comment (liza-veis) [inline src/services/http/index.ts]:
```suggestion
      headers.append(HttpHeader.AUTHORIZATION, token || '');
```

File: src/store/auth/actions.ts [hunk 38]
```diff
@@ -1,17 +1,38 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
 
-import { authApi } from 'src/services';
+import { authApi, storage } from 'src/services';
```
Comment (liza-veis) [inline src/store/auth/actions.ts]:
```suggestion
import { authApi, secureStorage } from 'src/services';
```

File: src/store/auth/actions.ts [hunk 39]
```diff
@@ -1,17 +1,38 @@
...
   ActionType.SIGN_IN,
   async (payload: ISignInPayload) => {
-    // TODO: implement login logic
-    const { user } = await authApi.signIn(payload);
+    const { token } = await authApi.signIn(payload);
 
-    return user;
+    await storage.setItem(SecureStorageKey.ACCESS_TOKEN, token);
+
+    return token;
```
Comment (liza-veis) [inline src/store/auth/actions.ts]:
```suggestion
    await secureStorage.setItem(SecureStorageKey.ACCESS_TOKEN, token);
    dispatch(loadCurrentUser());
```

File: src/store/auth/actions.ts [hunk 40]
```diff
@@ -1,17 +1,38 @@
...
+  // const { id, firstName, lastName, email } = await authApi.getCurrentUser();
+
+  const user = {
+    id: 1,
+    firstName: 'test',
+    lastName: 'test',
+    email: 'test@test.com',
+  };
+  return user;
+});
```
Comment (liza-veis) [inline src/store/auth/actions.ts]:
```suggestion
const loadCurrentUser = createAsyncThunk(
  ActionType.LOAD_CURRENT_USER,
  async () => {
    const token = await secureStorage.getItem(SecureStorageKey.ACCESS_TOKEN);

    if (!token) {
      return null;
    }

    return authApi.loadCurrentUser();
  }
);
```

File: src/store/auth/common.ts [hunk 41]
```diff
@@ -1,5 +1,6 @@
 enum ActionType {
   SIGN_IN = 'auth/sign-in',
+  SET_USER = 'auth/sign-in/set-user',
```
Comment (liza-veis) [inline src/store/auth/common.ts]:
```suggestion
  LOAD_CURRENT_USER = 'auth/load-current-user',
```

File: src/validation-schemas/login/login.ts [hunk 42]
```diff
@@ -0,0 +1,17 @@
+import * as yup from 'yup';
```
Comment (liza-veis) [inline src/validation-schemas/login/login.ts]: Let's move this file to validation-schemas/auth/login.ts and also rename this scheme to loginValidationSchema

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Comment by liza-veis** [comment]
We need to change the structure of the store a bit.

Let's write auth-slice like this (and then do the same for skill-slice): 
_store/auth/slice.ts_
```
const { reducer, actions } = createSlice({
  name: 'auth',
  initialState,
  reducers: {},
  extraReducers: builder => {
    builder.addCase(loadCurrentUser.fulfilled, (state, { payload }) => {
      state.user = payload;
    });
  },
});

export { reducer, actions };
```
Then in _store/auth/index.ts_
```
import { reducer, actions as sliceActions } from './slice';
import * as asyncActions from './actions';

const actions = {
  ...sliceActions,
  ...asyncActions,
};

export { reducer, actions };
```
_store/actions.ts_
```
export { actions as authActions } from './auth';
export { actions as skillActions } from './skill';
```
_store/root-reducer.ts_
```
import { reducer as auth } from './auth';
import { reducer as skill } from './skill';

export const rootReducer = {
  auth,
  skill,
};
```
Then we will use actions like this:
```
import { authActions } from 'src/store/actions';
...
dispatch(authActions.signIn(values));
```

File: src/components/form/form-password-input/form-password-input.tsx [hunk 43]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  return (
+    <View style={containerStyle}>
+      <Input
+        value={values[name]}
+        onChangeText={handleChange}
+        onBlur={handleBlur(name)}
+        secureTextEntry={secure}
+        outlineColor={!error ? AppColor.INPUT_BACKGROUND : AppColor.ERROR}
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-password-input/form-password-input.tsx]: You don't need to duplicate code from `from-input`. You can just add what you need and passthrough the rest.

File: src/screens/login/components/login-form/login-form.tsx [hunk 44]
```diff
@@ -0,0 +1,72 @@
+import React, { useCallback } from 'react';
+import { Alert, View } from 'react-native';
+import { Formik } from 'formik';
+
+import { styles } from '../../styles';
+import { MainButton } from 'src/components';
+import { defaultLoginPayload } from '../../common';
+import { authenticateUserValidationSchema } from 'src/validation-schemas';
+import { FormInput, FormPasswordInput } from 'src/components';
```
Comment (AleksandrTermenzhy) [inline src/screens/login/components/login-form/login-form.tsx]:
```suggestion
import { MainButton, FormInput, FormPasswordInput } from 'src/components';
import { defaultLoginPayload } from '../../common';
import { authenticateUserValidationSchema } from 'src/validation-schemas';
```

File: src/screens/login/login.tsx [hunk 45]
```diff
@@ -0,0 +1,41 @@
...
+
+import { LoginForm } from 'src/screens/login/components';
+import { styles } from './styles';
+import { AppRoute, AuthRoute, ButtonMode } from 'src/common/enums';
+import { MainButton } from 'src/components';
+
+const LoginScreen: React.FC = () => {
+  const navigation = useNavigation();
+
+  const handleNavigateTo = (route: AuthRoute | AppRoute) => {
```
Comment (AleksandrTermenzhy) [inline src/screens/login/login.tsx]: Why do we need this wrapper?

File: src/screens/login/styles.ts [hunk 46]
```diff
@@ -0,0 +1,38 @@
...
+import { AppColor } from 'src/common/enums';
+
+const styles = StyleSheet.create({
+  content: {
+    flex: 1,
+  },
+  container: {
+    flex: 1,
+    marginTop: '20%',
+    marginHorizontal: 27,
```
Comment (AleksandrTermenzhy) [inline src/screens/login/styles.ts]: What are these magic-numbers here and further?

File: src/store/auth/common.ts (hunk 41 above)
Comment (AleksandrTermenzhy) [reply src/store/auth/common.ts]: Where did you get that endpoint?

File: .vscode/launch.json [hunk 47]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (liza-veis) [inline .vscode/launch.json]: Remove this folder

File: src/common/types/user/authenticate-user.interface.ts (hunk 33 above)
Comment (liza-veis) [inline src/common/types/user/authenticate-user.interface.ts]: Remove this file

File: src/common/types/user/index.ts [hunk 48]
```diff
@@ -1 +1,2 @@
 export * from './user.interface';
+export * from './authenticate-user.interface';
```
Comment (liza-veis) [inline src/common/types/user/index.ts]:
```suggestion
```

File: src/components/form/form-password-input/form-password-input.tsx [hunk 49]
```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
```
Comment (liza-veis) [inline src/components/form/form-password-input/form-password-input.tsx:1]: Let's extend this component from form-input to not repeat the logic

File: src/components/input/input.tsx [hunk 50]
```diff
@@ -1,30 +1,28 @@
...
-);
+const Input: React.FC<InputProps> = ({
+  label,
+  placeholder,
+  style,
+  ...textInputProps
+}) => {
+  return (
+    <TextInput
+      style={[styles.inputContainer, style]}
```
Comment (liza-veis) [inline src/components/input/input.tsx]: Return ref and theme back

File: src/components/select/select.tsx [hunk 51]
```diff
@@ -24,7 +23,6 @@ const Select: React.FC<Props> = dropDownProps => {
       visible={open}
       showDropDown={handleShow}
       onDismiss={handleHide}
-      theme={theme}
```
Comment (Liza-Veis) [inline src/components/select/select.tsx:27]: Return theme back

File: src/validation-schemas/index.ts [hunk 52]
```diff
@@ -4,3 +4,5 @@ export * from './education';
 export * from './experience';
 export * from './language';
 export * from './location';
+export * from './auth';
+export * from './login';
```
Comment (liza-veis) [inline src/validation-schemas/index.ts]:
```suggestion
```

File: src/validation-schemas/login/login.ts (hunk 42 above)
Comment (liza-veis) [inline src/validation-schemas/login/login.ts]: Remove validation-schemas/login folder

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/screens/login/login.tsx [hunk 53]
```diff
@@ -0,0 +1,41 @@
...
+    navigation.navigate(route);
+  };
```
Comment (liza-veis) [inline src/screens/login/login.tsx]: It has wrong type now
```suggestion
  const handleSignUpPress = () => {
    navigation.navigate({
      name: AuthRoute.SIGN_UP as never,
      params: {} as never,
    });
  };
```

File: src/screens/register/register.tsx [hunk 54]
```diff
@@ -0,0 +1,33 @@
...
+import { AuthRoute, ButtonMode } from 'src/common/enums';
+
+type IRegisterScreenProps = Record<string, never>;
+
+const RegisterScreen: React.FC<IRegisterScreenProps> = () => {
+  const navigation = useNavigation();
+
+  const handleNavigateTo = (route: AuthRoute) => {
+    navigation.navigate(route);
+  };
```
Comment (liza-veis) [inline src/screens/register/register.tsx]:
```suggestion
  const handleSignInPress = () => {
    navigation.navigate({
      name: AuthRoute.SIGN_IN as never,
      params: {} as never,
    });
  };
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/components/form/form-password-input/form-password-input.tsx [hunk 55]
```diff
@@ -0,0 +1,35 @@
...
+import { FormInput } from 'src/components';
+import styles from '../styles';
+
+type FormInputProps = React.ComponentPropsWithoutRef<typeof FormInput>;
+
+type FormPasswordInputProps = Omit<FormInputProps, 'right'>;
+
+const FormPasswordInput: React.FC<FormPasswordInputProps> = ({
+  ...inputProps
+}) => {
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-password-input/form-password-input.tsx]:
```suggestion
const FormPasswordInput: React.FC<FormPasswordInputProps> = inputProps => {
```

File: src/components/form/form-password-input/form-password-input.tsx [hunk 56]
```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import { TextInput } from 'react-native-paper';
+
+import { FormInput } from 'src/components';
+import styles from '../styles';
+
+type FormInputProps = React.ComponentPropsWithoutRef<typeof FormInput>;
+
+type FormPasswordInputProps = Omit<FormInputProps, 'right'>;
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-password-input/form-password-input.tsx]: I don't think we need to extract `right` here. We might want to override it sometime in future.

File: src/components/form/form-password-input/form-password-input.tsx [hunk 57]
```diff
@@ -0,0 +1,35 @@
...
+    setSecure(!secure);
+  };
+
+  return (
+    <FormInput
+      {...inputProps}
+      secureTextEntry={secure}
+      right={
+        <TextInput.Icon
+          name="eye"
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-password-input/form-password-input.tsx]: We should also change icon on touch to reflect open/close states

File: src/screens/index.ts [hunk 58]
```diff
@@ -3,9 +3,6 @@ export * from './mentee-home';
...
 export * from './opportunities';
 export * from './profile';
+export * from './login';
+export * from './register';
 export * from './create-skill';
-export * from './onboarding';
-export * from './add-experience';
-export * from './add-education';
-export * from './add-language';
-export * from './add-location';
```
Comment (AleksandrTermenzhy) [inline src/screens/index.ts]: Why do we need to delete these?

File: src/screens/login/components/login-form/login-form.tsx [hunk 59]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Alert, View } from 'react-native';
+import { Formik } from 'formik';
+
+import { styles } from '../../styles';
+import { MainButton } from 'src/components';
+import { defaultLoginPayload } from '../../common';
+import { loginValidationSchema } from 'src/validation-schemas';
+import { FormInput, FormPasswordInput } from 'src/components';
```
Comment (AleksandrTermenzhy) [inline src/screens/login/components/login-form/login-form.tsx]:
```suggestion
import { defaultLoginPayload } from '../../common';
import { loginValidationSchema } from 'src/validation-schemas';
import { FormInput, FormPasswordInput, MainButton } from 'src/components';
```

File: src/screens/register/register.tsx (hunk 54 above)
Comment (AleksandrTermenzhy) [inline src/screens/register/register.tsx]: Why do we need this wrapper if we still use it just like `navigate()`?

**Review by AleksandrTermenzhy — APPROVED** [review]

File: src/screens/login/login.tsx [hunk 60]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { View, Image, Text } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+
+import { LoginForm } from 'src/screens/login/components';
+import { styles } from './styles';
```
Comment (yevhenprybytok) [inline src/screens/login/login.tsx]: Imports order should be respected

File: src/screens/login/components/login-form/login-form.tsx [hunk 61]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Alert, View } from 'react-native';
+import { Formik } from 'formik';
+
+import { styles } from '../../styles';
```
Comment (yevhenprybytok) [inline src/screens/login/components/login-form/login-form.tsx]: Imports order here should be respected

**Review by yevhenprybytok — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]
